modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/DiscoveryDataPacket.java

```diff
@@ -39,6 +39,8 @@
  * and allows convenient way of converting it to and from {@link DiscoveryDataBag} objects.
  */
 public class DiscoveryDataPacket implements Serializable {
+    /** Local file header signature(read as a little-endian number). */
+    private static int ZIP_HEADER_SIGNATURE = 0x04034b50;
```

Comment:
Are you sure that correct uncompressed message cannot be generated which will start from this signature?